Model/Service/DumpCmsDataService.php

```diff
@@ -78,28 +77,17 @@ private function write(WriteInterface $writeDirectory, string $filePath, string
         $stream->close();
     }
 
-    private function getBlocksMapping(): array
-    {
-        $blocksMapping = [];
-        $searchCriteria = $this->criteriaBuilder;
-        $blocksList = $this->blockRepository->getList($searchCriteria->create());
-        $blocks = $blocksList->getItems();
-        foreach ($blocks as $block) {
-            $blocksMapping[$block->getId()] = $block->getIdentifier();
-        }
-
-        return $blocksMapping;
-    }
-
     private function replaceBlockIds(string $content): string
     {
         preg_match_all('/block_id=\"([0-9]+)\"/', $content, $blockIds);
         if (isset($blockIds[1])) {
             foreach ($blockIds[1] as $blockId) {
-                if (isset($this->blocksMapping[$blockId])) {
-                    $identifier = $this->blocksMapping[$blockId];
-                    $content = str_replace("block_id=\"$blockId\"", "block_id=\"$identifier\"", $content);
+                if (!isset($this->blocksMapping[$blockId])) {
```

Comment:
Better, but this approach now pulls from DB for every new block ID. I suggest pulling the data per replaceBlockIds() call. After `if (isset($blockIds[1])) {` you have a collection of $blockIds, use that to pull all of Ids at once. This way, we pull from DB only once per CMS block.